src/coreComponents/fileIO/logPart/LogPart.hpp

```diff
@@ -0,0 +1,140 @@
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file LogPart.hpp
+ */
+
+#ifndef GEOS_COMMON_SECTION_HPP
+#define GEOS_COMMON_SECTION_HPP
+
+#include "common/DataTypes.hpp"
+#include "common/format/Format.hpp"
+
+namespace geos
+{
+
+/**
+ * @brief Class for displaying different steps of simulation
+ */
+class LogPart
+{
+public:
+
+  /**
+   * @brief Construct a new LogPart
+   * @param m_logPartTitle The logPart title
+   */
+  LogPart( string_view m_logPartTitle );
+
+  /**
+   * @brief Add a description to the logPart by concatening a description name and descriptions values.
+   * @param name The description name
+   * @param args Descriptions values to be concatened.
+   * Descriptions values can be be any types and will be aligned
+   */
+  template< typename ... Args >
+  void addDescription( string const & name, Args const & ... args );
+
+  /**
+   * @brief Add a description to the logPart
+   * @param description The string value of the description
+   */
+  void addDescription( string const & description );
+
+  /**
+   * @brief Set the minimal width of a row
+   * @param minWidth The minimal width of the table
+   */
+  void setMinWidth( size_t const & minWidth );
+
+  /**
+   * @brief Draw the first part of the logPart. It include the title and optionnaly, the end description(s).
+   * @param os An output stream (by default, std::cout)
+   */
+  void begin( std::ostream & os = std::cout );
+
+  /**
+   * @brief Draw the last part of the logPart. It include the title
+   * @param oss An output stream (by default, std::cout)
+   */
+  void end( std::ostream & oss = std::cout );
+
+private:
+
+  /**
+   * @brief Build a description from the name and description values
+   * @param name The decription name
+   * @param decriptionsValues The description values
+   */
+  void formatDescriptions();
+
+  /**
+   * @brief Constructs the string logPart title of the log part.
+   */
+  string buildTitlePart();
+
+  /**
+   * @brief Constructs the string logPart descriptions of the log part.
+   */
+  string buildDescriptionPart();
+
+  /**
+   * @brief Clear all buffer
+   */
+  void clear();
+
+  /// Vector containing formatted description
+  std::vector< string > m_formattedDescriptions;
+  /// Name of the description, formatted to be : [Name] : [Values1]\n[Values2]
+  std::vector< string > m_descriptionNames;
+  /// Values in the descrpption the description
+  std::vector< std::vector< string > > m_descriptionValues;
+
+  /// title of logPart
+  string m_logPartTitle;
+  /// Title footer string
+  string m_footerTitle;
+  /// min width of logPart length
+  size_t m_rowMinWidth = 70;
+  /// logPart length
+  size_t m_logPartWidth;
```

Comment:
```suggestion
  struct Description {
    /// Title footer string
    string m_title;

    /// Name of the description, formatted to be : [Name] : [Values1]\n[Values2]
    std::vector< string > m_descriptionNames;
    /// Values in the descrpption the description
    std::vector< std::vector< string > > m_descriptionValues;
    /// Vector containing the descriptions formatted by formatDescriptions()
    std::vector< string > m_formattedDescriptionLines;

    /// logPart length
    size_t m_logPartWidth;
    
    void formatDescriptions();
  };

  /// min width of logPart length
  size_t m_rowMinWidth = 70;

  Description startDesc;
  Description endDesc;
```

And so, you should be able to easily restore the `addEndDescription()` methods, which should be more user friendly from my point of view.

I'm not sure that `void formatDescriptions();` should be relocated here (and that its signature is complete), but if the code is well structured, it should be possible.

Reply:
i've done a lot of refafac with your proposal 👨‍🍳